src/services/sync/strategies/LDAPSyncer.js

```diff
@@ -32,111 +39,152 @@ class LDAPSyncer extends Syncer {
+				}
+			}
+		} catch (err) {
+			this.logger.error('Uncaught LDAP sync error', { error: err, systemId: this.system._id, syncId: this.syncId });
+		}
+		return schoolList;
+	}
+
+	async getUserData(school) {
+		this.logInfo(`Getting users for school ${school.name}`, { syncId: this.syncId });
+		const ldapUsers = await this.app.service('ldap').getUsers(this.system.ldapConfig, school);
+		this.logInfo(`Creating and updating ${ldapUsers.length} users for school ${school.name}`, { syncId: this.syncId });
+
+		const bulkSize = 1000; // 5000 is a hard limit because of definition in user model
+
+		for (let i = 0; i < ldapUsers.length; i += bulkSize) {
+			const ldapUserChunk = ldapUsers.slice(i, i + bulkSize);
+			for (const ldapUser of ldapUserChunk) {
+				try {
+					const userData = {
+						action: LDAP_SYNC_ACTIONS.SYNC_USER,
+						syncId: this.syncId,
+						data: {
+							user: {
+								firstName: ldapUser.firstName,
+								lastName: ldapUser.lastName,
+								systemId: this.system._id,
+								schoolDn: school.ldapSchoolIdentifier,
+								email: ldapUser.email,
+								ldapDn: ldapUser.ldapDn,
+								ldapId: ldapUser.ldapUUID,
+								roles: ldapUser.roles,
+							},
+							account: {
+								ldapDn: ldapUser.ldapDn,
+								ldapId: ldapUser.ldapUUID,
+								username: `${school.ldapSchoolIdentifier}/${ldapUser.ldapUID}`.toLowerCase(),
+								systemId: this.system._id,
+								schoolDn: school.ldapSchoolIdentifier,
+								activated: true,
+							},
+						},
+					};
+					this.syncQueue.sendToQueue(userData, {});
+				} catch (err) {
+					this.logError(`User creation error for ${ldapUser.firstName} ${ldapUser.lastName} (${ldapUser.email})`, {
+						err,
+						syncId: this.syncId,
+					});
+				}
+			}
+		}
+	}
+
+	async getClassData(school) {
+		this.logInfo(`Getting classes for school ${school.name}`, { syncId: this.syncId });
+		const classes = await this.app.service('ldap').getClasses(this.system.ldapConfig, school);
+		this.logInfo(`Creating and updating ${classes.length} classes for school ${school.name}`, { syncId: this.syncId });
+		for (const ldapClass of classes) {
+			try {
+				this.pushClassData(ldapClass, school);
+			} catch (err) {
+				this.logError('Cannot create synced class', { error: err, ldapClass, syncId: this.syncId });
+			}
+		}
+	}
+
+	pushClassData(data, school) {
+		const classData = {
+			action: LDAP_SYNC_ACTIONS.SYNC_CLASSES,
+			syncId: this.syncId,
+			data: {
+				name: data.className,
+				systemId: this.system._id,
+				schoolDn: school.ldapSchoolIdentifier,
+				nameFormat: 'static',
+				ldapDN: data.ldapDn,
+				year: school.currentYear,
+				uniqueMembers: data.uniqueMembers,
+			},
+		};
+		if (!Array.isArray(classData.uniqueMembers)) {
+			// if there is only one member, ldapjs doesn't give us an array here
+			classData.uniqueMembers = [classData.uniqueMembers];
+		}
+		this.syncQueue.sendToQueue(classData, {});
```

Comment:
error handling  missing

Reply:
coude you describe it a bit more, what handling are we missing?